internal/pkg/scorecard/scorecard.go

```diff
@@ -171,37 +171,51 @@ func runTests() ([]scapiv1alpha1.ScorecardOutput, error) {
 			return nil, err
 		}
 
-		// Create a temporary CR manifest from metadata if one is not provided.
-		crJSONStr, ok := csv.ObjectMeta.Annotations["alm-examples"]
-		if ok && viper.GetString(CRManifestOpt) == "" {
-			var crs []interface{}
-			if err = json.Unmarshal([]byte(crJSONStr), &crs); err != nil {
-				return nil, err
-			}
-			// TODO: run scorecard against all CR's in CSV.
-			cr := crs[0]
-			crJSONBytes, err := json.Marshal(cr)
-			if err != nil {
-				return nil, err
-			}
-			crYAMLBytes, err := yaml.JSONToYAML(crJSONBytes)
-			if err != nil {
-				return nil, err
-			}
-			crFile, err := ioutil.TempFile("", "cr.yaml")
-			if err != nil {
-				return nil, err
-			}
-			if _, err := crFile.Write(crYAMLBytes); err != nil {
-				return nil, err
-			}
-			viper.Set(CRManifestOpt, crFile.Name())
-			defer func() {
-				err := os.Remove(viper.GetString(CRManifestOpt))
+		logCRMsg := false
+		if crMans := viper.GetStringSlice(CRManifestOpt); len(crMans) == 0 {
+			// Create a temporary CR manifest from metadata if one is not provided.
+			if crJSONStr, ok := csv.ObjectMeta.Annotations["alm-examples"]; ok {
+				var crs []interface{}
+				if err = json.Unmarshal([]byte(crJSONStr), &crs); err != nil {
+					return nil, err
+				}
+				// TODO: run scorecard against all CR's in CSV.
+				cr := crs[0]
+				logCRMsg = len(crs) > 1
+				crJSONBytes, err := json.Marshal(cr)
 				if err != nil {
-					log.Errorf("Could not delete temporary CR manifest file: (%v)", err)
+					return nil, err
 				}
-			}()
+				crYAMLBytes, err := yaml.JSONToYAML(crJSONBytes)
+				if err != nil {
+					return nil, err
+				}
+				crFile, err := ioutil.TempFile("", "*.cr.yaml")
```

Comment:
Why is the file `*.cr.yaml` now?

Reply:
The temp file names were in an annoying format before :grin:. This format uses `.cr.yaml` as a file extension.